src/NLog.Raven/RavenTarget.cs

```diff
@@ -122,11 +128,22 @@ protected override void Write(IList<AsyncLogEventInfo> logEvents)
             {
                 using (var bulkInsert = _documentStore.BulkInsert())
                 {
-                    for (int i = 0; i < logEvents.Count; ++i)
+                    foreach (var log in logEvents)
                     {
-                        var logEvent = logEvents[i].LogEvent;
+                        var logEvent = log.LogEvent;
                         var logEntry = CreateLogEntry(logEvent);
-                        bulkInsert.Store(logEntry);
+                        if (ExpirationOffsetInDays > 0)
+                        {
+                            var expiry = DateTime.UtcNow.AddDays(ExpirationOffsetInDays);
+                            bulkInsert.Store(logEntry, new MetadataAsDictionary
```

Comment:
Could one make a single `MetadataAsDictionary`-instance, and use the same instance for all logEntries?

Reply:
Now using one instance for all entries.